### PR TITLE
Change num_seqs type from int to torch.Tensor

### PR DESCRIPTION
**Pull request by vanbasten23**
In vLLM, I see excessive dynamo recompile ([output](https://gist.github.com/vanbasten23/553dbe3d58d8c1530f5daa10155634eb)), for example `L['attn_metadata'].num_seqs == 10 `. Per https://pytorch.org/docs/main/torch.compiler_troubleshooting.html#wrapping-constants-with-tensors, we can wrap the int with tensors to reduce the compilation. So this PR changes the num_seqs type from int to torch.Tensors.

### CODE REVIEW

File: torch_xla/experimental/custom_kernel.py [hunk 1]
```diff
@@ -773,13 +773,14 @@ def ragged_paged_attention(
...
     cu_q_lens,  # i32[num_tokens + 1]
-    num_seqs,  # int
+    num_seqs,  # i32[]
     num_kv_pages_per_block,
     num_queries_per_block,
     use_kernel=True,
     # TODO(jevinjiang, xiowei): add attn_logits_soft_cap.
     # attn_logits_soft_cap: float | None = None,
 ):  # [batch_size, query_len, num_heads, head_dim]:
+  num_seqs = num_seqs.item()
```
Comment (yaochengji) [inline torch_xla/experimental/custom_kernel.py:783]: Why is there a `.item()` here?

Comment (vanbasten23) [reply torch_xla/experimental/custom_kernel.py:783]: because the argument `num_seqs` has type torch.Tensor, but the jax kernel needs to take an int type. So I need to convert the torch.Tensor to int.
I tried to also make the kernel to also take a jnp.array [here](https://github.com/pytorch/xla/blob/1acc987aec541978cd52ea77a027f9028d8148d3/torch_xla/experimental/pallas_kernels/ragged_paged_attention_kernel.py#L736) but it's not trivial to make the change in the kernel.

Comment (yaochengji) [reply torch_xla/experimental/custom_kernel.py:783]: There will be a tiny graph here. What will happen if we combine this with torch.compile?

Comment (vanbasten23) [reply torch_xla/experimental/custom_kernel.py:783]: You meant a graph break? The correctness tests seems fine and it has a torch.compile test. But I'm not sure perf-wise what will happen.

Comment (vanbasten23) [reply torch_xla/experimental/custom_kernel.py:783]: Each scheduler iteration will have different num_seqs

Comment (vanbasten23) [reply torch_xla/experimental/custom_kernel.py:783]: I did some experiment https://paste.googleplex.com/4746446904229888 and it seems `.item()` won't create a graph break. I attached the HLO in the link above

**Review by yaochengji — APPROVED** [review]
LGTM, thanks!
